samples/can/src/main.c

```diff
@@ -0,0 +1,294 @@
+/*
+ * Copyright (c) 2018 Alexander Wachter
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr.h>
+#include <kernel.h>
+#include <misc/printk.h>
+#include <device.h>
+#include <can.h>
+#include <gpio.h>
+
+#define CAN_DEV "CAN_1"
+#define GPIO_LED_DEV "GPIOC"
+#define GPIO_BUTTON_DEV "GPIOA"
+#define PIN_USER_BUTTON 0
+#define PIN_LED_1 6
+#define PIN_LED_2 7
+#define PIN_LED_3 8
+#define PIN_LED_4 9
+#define TX_THREAD_STACK_SIZE 512
+#define LED_THREAD_STACK_SIZE 512
+#define RX_STR_THREAD_STACK_SIZE 512
+#define TX_THREAD_PRIORITY 2
+#define LED_MSG_ID (0x10)
+#define BUTTON_MSG_ID (0x01)
+#define STR_MSG_ID (0x12345)
+#define LOOPBACK 1
+
+K_THREAD_STACK_DEFINE(tx_thread_stack, TX_THREAD_STACK_SIZE);
+K_THREAD_STACK_DEFINE(led_thread_stack, LED_THREAD_STACK_SIZE);
+K_THREAD_STACK_DEFINE(rx_str_thread_stack, RX_STR_THREAD_STACK_SIZE);
+struct k_thread tx_thread_data;
+struct k_thread led_thread_data;
+struct k_thread rx_str_thread_data;
+struct k_sem tx_sem;
+static struct gpio_callback gpio_cb;
+CAN_DEFINE_MSGQ(led_msgq, 2);
+CAN_DEFINE_MSGQ(str_msgq, 5);
+
+void tx_irq_callback(u32_t error_flags)
+{
+	printk("Callback! error-code: %d\n", error_flags);
+}
+
+void button_callback(struct device *port,
+				   struct gpio_callback *cb, u32_t pins)
+{
+	k_sem_give(&tx_sem);
+}
+
+void send_string(char* string, struct device *can_dev)
+{
+	struct can_msg msg;
+	msg.ext_id = STR_MSG_ID;
+	msg.id_type = can_extended_identifier;
+	msg.dlc = 0;
+	msg.rtr = can_dataframe;
+	for(int str_len = strlen(string); str_len;)
```

Comment:
do "for (str_len = ...." instead here, so space after "for" and place variable declaration at the start of the func